internal/replicasets/replicasets.go

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package replicasets
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetReplicasetByDeployment(cli client.Client, ctx context.Context, deployment appsv1.Deployment) ([]appsv1.ReplicaSet, error) {
```

Comment:
I'd merge this function into `internal/podlist` mimcing what the `DeploymentByOwnerReference` is doing (bar the necessary adjustments, so perhaps `ReplicaSetByDeployment`?)

admittedly `podlist` is not the best location, but a new `replicasets` package is not obviously good so I'd keep everything packed so we can fix all the mess in one go in the future

Reply:
I was thinking similarly but I didn't see ReplicaSetByDeployment fits under podList just because of the naming of this folder. It conveys that the functionality presented there lists pod under other controllers. so I created a new folder for replicaset (although there is a high chance we'll not gonna use it for other methods in the near future as we concentrate on pods mainly).
Anyhow, no big deal of moving it under podList, so will do so.

Reply:
let me know what do you think if we need to rename podList to something else perhaps findby

Reply:
I concur `podlist` is not the best fit for this function and considering only this change moving the new code here is actually a pessimization. At project level there's no clear direction here so I prefer to keep all the related code packed instead of experimenting with independent solutions. I'd rather refactor everything in one go